### PR TITLE
adding fp16 support for topk cuda kernel

### PR DESCRIPTION
**Pull request by duli2012**
**Description**: Describe your changes.
adding fp16 support for topk cuda kernel

### CODE REVIEW

File: onnxruntime/core/providers/cuda/math/topk.cc [hunk 1]
```diff
@@ -95,9 +95,9 @@ Status TopK<inputk>::ComputeInternal(OpKernelContext* ctx) const {
   if (IS_PRIM_TYPE(int16_t)) return TOPKIMPL(int16_t);
   if (IS_PRIM_TYPE(int32_t)) return TOPKIMPL(int32_t);
   if (IS_PRIM_TYPE(int64_t)) return TOPKIMPL(int64_t);
+  if (IS_PRIM_TYPE(MLFloat16)) return TOPKIMPL(MLFloat16);
```
Comment (hanbitmyths) [inline onnxruntime/core/providers/cuda/math/topk.cc:98]: Please, check if we have a unit test for supporting types.

Comment (hariharans29) [inline onnxruntime/core/providers/cuda/math/topk.cc:98]: nits: 
1) Maybe the floating types can come ahead of the integer types in the conditional branching order ? 
2) A test would be nice for the new type supported.

Comment (yuslepukhin) [inline onnxruntime/core/providers/cuda/math/topk.cc:98]: >L(MLFloat16); [](start = 45, length = 13)

Does the kernel handle float16?

Comment (duli2012) [reply onnxruntime/core/providers/cuda/math/topk.cc:98]: added

Comment (duli2012) [reply onnxruntime/core/providers/cuda/math/topk.cc:98]: fixed

**Comment by hanbitmyths** [comment]
A unit test failed on CPU. It seems TopK CPU kernel doesn't support FP16?

**Comment by duli2012** [comment]
> A unit test failed on CPU. It seems TopK CPU kernel doesn't support FP16?

Yes. I disabled the test for cpu EP to unblock this PR. if we need fp16 on cpu, we can do it in another PR.

**Review by hanbitmyths — APPROVED** [review]

**Pull request merged**